File: 0792-binary-search/0792-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:

        def binary(start:int,end:int):
            if start>=end:
                return -1
            m=(start+end)//2
            if nums[m]==target:
                return m
            elif nums[m]<target:
                return binary(m+1,end)
            else:
                return binary(start,m)
            

        return binary(0,len(nums))


        

File: 0792-binary-search/0792-binary-search.ts
function search(nums: number[], target: number): number {
    let n=nums.length
    let low=0
    let high=n-1
    while(low<=high){
       let mid=Math.floor((low+high)/2)
       console.log(low,mid,high)
        if(nums[mid]==target)return mid
        else if(nums[mid]<target)low=mid+1
        else high=mid-1
    }
    return -1
    
};